docs/asf/pyanalysis.rst

```diff
@@ -0,0 +1,298 @@
+
+
+.. code-block :: bash
+
+    export PYTHONPATH=/usr/bin/python:/<INSTALL_PATH>/lib/python<PYTHON_VERSION>/site-packages/
+    export PATH=/usr/bin:/<INSTALL_PATH>/bin:/<INSTALL_PATH>/sbin::$PATH
+
+Then you can imitate the Grafana query to call your analysis module using a python script such as:
+
+.. code-block :: python
+
+    #!/usr/bin/python3
+
+    import time,sys
+    from sosdb import Sos
+    from grafanaFormatter import DataFormatter
+    from table_formatter import table_formatter
+    from time_series_formatter import time_series_formatter
+    from dsosTemplate import dsosTemplate
+
+    sess = Sos.Session("/<DSOS_CONFIG_PATH>/config/dsos.conf")
+    cont = '<PATH_TO_DATABASE>'
+    cont = sess.open(cont)
+
+    model = dsosTemplate(cont, time.time()-300, time.time(), schema='meminfo', maxDataPoints=4096)
+
+    x = model.get_data(['Active'])
+
+    #fmt = table_formatter(x)
+    fmt = time_series_formatter(x)
+    x = fmt.ret_json()
+    print(x)
+
+* The ``model.get_data`` is where you can define the type of metrics to collect (in this case it is "Active"), what filters and extra parameters you want to add to your query. The syntax is as follows: ``(['<metric>'], filters=['job_id>0'], params='<variable>')``
+
+* If you would like to query all metrics then replace ``Active`` with ``*``.
+* To query a specific job_id: set ``job_id`` to you job_id with ``==``.
+* To query from a specific time range: update the start time, ``time.time()-300`` and end time, ``time.time()`` to an epoch timestamp.
+* To add a string metric, filter or parameter, you must include a double quote, ``"``, before and after the string (i.e. ``filters=['user=="myusername"']``)
+
+.. note::
+
+	The ``params`` can be any number or string that you want to define in your analysis module to better manage, output or analyze the data. For example, you can program your module to return specific analyses such as the average with ``params='analysis=average'`` by parsing the arguement, using ``if`` statements to determine what analysis to apply to the data and, to make things cleaner, a function to perform these calculations in.
+/t
+Required Scripts
+////////////////
+The following scripts are needed to run the python analysis module. If these python scripts or modules **do not exist on your system and you have no way of accessing them** then please continue. Otherwise, you can skip this section
+
+**If you do not have access to these existing scripts** then please create them in the same directory as your python analysis module.
+
+.. note::
+
+  If Grafana and SOS are installed on your system then please ask your system administator where these files reside on the system so that you can copy them to your home directory.
+
+grafanaFormatter:
+
+.. code:: RST
```

Comment:
Shouldn't this be "python" instead of RST? The same for number of the following code blocks.

Reply:
Yes, you are correct. It has been fixed, thanks for the catch!